### DISCUSSION TITLE
Condition "read data" checking if data already exist

### DISCUSSION THREAD

**Discussion by BrunoUY** [directus/directus#14443] [Archive]
Hi!

It's my first time trying to use it flows, and I can´t get it to do it right.

I have a **collection** with a field **asset_symbol**

### I have this flow
![imagen](https://user-images.githubusercontent.com/4197644/179123749-6ffb0439-0471-44e0-9e44-76fced729d4e.png)

------------------

### This event hook
![imagen](https://user-images.githubusercontent.com/4197644/179123787-0b79d43a-e60c-4d3a-b43f-6f41f3663d8f.png)

------------------

### On the operation I have
![imagen](https://user-images.githubusercontent.com/4197644/179123812-3534d827-5c20-45f6-8734-cd6b582a5112.png)

------------------

### When I run it, I get the log as this:
![imagen](https://user-images.githubusercontent.com/4197644/179123916-d3739567-c68b-4166-abc7-d4db2d35f51e.png)

I don´t fully understad why is not working, and what is about the status code 403, I tried several configurations but nothing seems to work.

any idea on this one?

thank you a lot!,

pd: flows looks awesome!

**Comment by fitu86** [comment]
It is because it did not find the image you are looking for, error 200 is that it was found and 403 is that it did not.

**Reply by BrunoUY** [reply]
Hi,

Thank you for the response

I have checked wih an asset that already exist and it shows the same error, also the flow it does not go to the correct "log to console"

I check in the docs, and looks like 403 FORBIDDEN is
![imagen](https://user-images.githubusercontent.com/4197644/179227381-1c86ae5e-47a5-4be9-87e9-b4adbc44ea30.png)

I'm doing something wrong in the permission settingson the flow?

**Comment by BrunoUY** [comment]
I have tested another Query, maybe that was wrong

```
{
  "filter": {
    "asset_symbol": {
      "_eq": "{{$last.payload.asset_symbol}}"
    }
  }
}
```

but I keep receiving the same error,

any ideas ?

**Comment by rijkvanzanten** [comment]
In the operation configuration, you have `Permissions` set to "From Trigger", meaning that it will use the read permissions from the user who triggered the flow. Try setting that to "Full Access" temporarily to see if that was the cause of the permissions problem 🙂

**Reply by BrunoUY** [reply]
thanks for the answer

it is not working.

**I have tried to delete the flow or inactivate it, to create a new one, but I can´t'**

the error is 

```
{
  "errors": [
    {
      "message": "delete from `directus_operations` where `id` in ('73888d01-cdc2-4b53-871b-a883fe84b3a5', '8d4c70fc-d598-4208-bd85-44579aa796fe', '8ffee19f-571f-4323-b228-b78f2c5b504f') - ER_ROW_IS_REFERENCED_2: Cannot delete or update a parent row: a foreign key constraint fails (`asset_screener_v2`.`directus_operations`, CONSTRAINT `directus_operations_resolve_foreign` FOREIGN KEY (`resolve`) REFERENCES `directus_operations` (`id`))",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```

Does it look like a directus internal ?

What can I do to delete the flow?

**Reply by rijkvanzanten** [reply]
I think you already opened another discussion about this, so lets keep this there 👍🏻  https://github.com/directus/directus/discussions/14463
